src/main/java/org/shanerx/tradeshop/utils/debug/Debug.java

```diff
@@ -28,58 +28,139 @@
+     * @implNote This method is called by {@link #reload()} to load the debug level. It should add enums from DebugLevels to the {@link HashSet} specified by debugDestination to enable debugging for each item.
+     * @var debugDestination The {@link HashSet} to add the debug levels to
+     * @var settingToReadFrom The {@link Setting} to read the debug level from
+     */
+    public abstract boolean loadDebugLevel(Setting settingToReadFrom, HashSet<DebugLevels> debugDestination);
 
-        if (decimalDebugLevel > 0) {
-            Bukkit.getLogger().log(Level.INFO, PREFIX.replace("%level%", "") + "Debugging enabled!");
-            Bukkit.getLogger().log(Level.INFO, PREFIX.replace("%level%", "") + "Decimal Debug level: " + decimalDebugLevel);
-            Bukkit.getLogger().log(Level.INFO, PREFIX.replace("%level%", "") + "Debug levels: " + binaryDebugLevel);
-        }
+    public String getFormattedPrefix(DebugLevels debugLevel) {
+        return PREFIX.replace("%level%", debugLevel.getPrefix());
     }
 
     public void log(String message, DebugLevels level) {
-        if (level.getPosition() > 0 && binaryDebugLevel.charAt(level.getPosition() - 1) == '1') {
-            Bukkit.getLogger().log(level.getLogLevel(), PREFIX.replace("%level%", level.getPrefix()) + message);
+        log(message, level, null);
+    }
+
+    public void log(String message, DebugLevels level, String positionalNote) {
+        StringBuilder messageBuilder = new StringBuilder();
+
+        if (debugToConsole.contains(level)) {
+            message = PREFIX.replace("%level%", level.getPrefix()) + message;
         } else if (level == DebugLevels.DISABLED) {
-            Bukkit.getLogger().log(level.getLogLevel(), PREFIX.replace(" Debug%level%", "") + message);
+            message = PREFIX.replaceAll("( Debug.%level%)", "") + message;
         } else if (level.getPosition() < 0) {
-            Bukkit.getLogger().log(level.getLogLevel(), PREFIX.replace("%level%", level.getPrefix()) + message);
+            message = PREFIX.replace("%level%", level.getPrefix()) + message;
         }
+
+        if (positionalNote != null && !positionalNote.isEmpty()) {
+            messageBuilder.append("{ ").append(level.getPrefix()).append(" }\n");
+        }
+
+        messageBuilder.append(message);
+
+        if (debugToConsole.contains(level)) logToConsole(level.getLogLevel(), messageBuilder.toString());
+        if (debugToFile.contains(level)) logToFile(level.getLogLevel(), messageBuilder.toString());
     }
 
-    public String getFormattedPrefix(DebugLevels debugLevel) {
-        return PREFIX.replace("%level%", debugLevel.getPrefix());
+    private void logToConsole(Level level, String message) {
+        Bukkit.getLogger().log(level, message);
+    }
+
+    private void logToFile(Level level, String message) {
+        Bukkit.getLogger().log(level, "\n----- LOGTOFILE -----" + message + "\n----- END LOGTOFILE -----\n"); //TODO: Add file logging
```

Comment:
This is still missing right?

Reply:
Yes, just pushing to console with the different format